listener.go

```diff
@@ -58,11 +57,12 @@ func NewListener(l net.Listener, storage TLSStorage, caCert *x509.Certificate, c
 			FilterCN:            allowDefaultSANs(config.SANs, config.FilterCN),
 			ExpirationDaysCheck: config.ExpirationDaysCheck,
 		},
-		Listener:  l,
-		storage:   &nonNil{storage: storage},
-		sans:      config.SANs,
-		maxSANs:   config.MaxSANs,
-		tlsConfig: config.TLSConfig,
+		Listener:     l,
+		certIsSetSig: make(chan struct{}),
```

Comment:
Nit: we tend to call these ready channels in other projects, what about:
```suggestion
		certReady: make(chan struct{}),
```